File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/showcase/ShowCaseUtils.java
package cz.cvut.fel.matyapav.afandroid.showcase;

import android.app.Activity;
import android.content.SharedPreferences;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;

import java.util.HashMap;

import cz.cvut.fel.matyapav.afandroid.R;

/**
 * Created by Pavel on 20.02.2016.
 */
public class ShowCaseUtils {

    public static String PREFS_NAME = "preferences";
    public static int PRIVATE_MODE = 0;

    public static void setUserInPreferences(Activity activity, String username, String password){
        SharedPreferences mySharedPreferences= activity.getSharedPreferences(PREFS_NAME, PRIVATE_MODE); //0 is private mode
        SharedPreferences.Editor editor= mySharedPreferences.edit();
        editor.putString("username",username);
        editor.putString("password", password);
        editor.commit();
    }

    public static void clearUserInPreferences(Activity activity){
        SharedPreferences mySharedPreferences = activity.getSharedPreferences(PREFS_NAME, PRIVATE_MODE);
        SharedPreferences.Editor editor= mySharedPreferences.edit();
        editor.remove("username");
        editor.remove("password");
        editor.commit();
    }

    public static HashMap<String, String> getUserCredentials(Activity activity){
        SharedPreferences mySharedPreferences = activity.getSharedPreferences(PREFS_NAME, PRIVATE_MODE);
        String username = mySharedPreferences.getString("username", null);
        String password = mySharedPreferences.getString("password", null);
        if(username != null && password != null){
            HashMap<String, String> result = new HashMap<>();
            result.put("username", username);
            result.put("password", password);
            return result;
        }
        return null;
    }

    public static void refreshCurrentFragment(FragmentActivity activity){
        FragmentManager fragmentManager = activity.getSupportFragmentManager();
        Fragment current = fragmentManager.findFragmentById(R.id.mainLayout);
        fragmentManager.beginTransaction().detach(current).attach(current).commit();
    }
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/showcase/utils/ShowcaseUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml.Controls;

namespace AFWindowsPhone.showcase
{
    class ShowcaseUtils
    {

        public static void setUserInPreferences(String username, String password)
        {
            var localSettings = Windows.Storage.ApplicationData.Current.LocalSettings;
            localSettings.Values["username"] = username;
            localSettings.Values["password"] = password;
        }

        public static void clearUserInPreferences()
        {
            var localSettings = Windows.Storage.ApplicationData.Current.LocalSettings;
            localSettings.Values.Remove("username");
            localSettings.Values.Remove("password");
        }

        public static Dictionary<String, String> getUserCredentials()
        {
            var localSettings = Windows.Storage.ApplicationData.Current.LocalSettings;
            String username = (String) localSettings.Values["username"];
            String password = (String) localSettings.Values["password"];
            if (username != null && password != null)
            {
                Dictionary<String, String> result = new Dictionary<String, String>();
                result.Add("username", username);
                result.Add("password", password);
                return result;
            }
            return null;
        }

        public static String getUserLogin()
        {
            var localSettings = Windows.Storage.ApplicationData.Current.LocalSettings;
            String username = (String) localSettings.Values["username"];
            return username;
        }

    }
}
